#! /usr/bin/env python

PACKAGE='lead_follow'

from dynamic_reconfigure.msg import SensorLevels
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#       Name                Type       Lvl    Description                                   Default     Min     Max
gen.add("x_min",                    double_t,  0,     "Region of interest",                                -100.0,     -1000.0, 0.0)
gen.add("x_max",                    double_t,  0,     "Region of interest",                                100.0,      0.0,     1000.0)
gen.add("y_min",                    double_t,  0,     "Region of interest",                                -100.0,     -1000.0, 0.0)
gen.add("y_max",                    double_t,  0,     "Region of interest",                                100.0,      0.0,     1000.0)
gen.add("z_min",                    double_t,  0,     "Region of interest",                                -100.0,     -1000.0, 0.0)
gen.add("z_max",                    double_t,  0,     "Region of interest",                                3.0,        0.0,     1000.0)
gen.add("voxel_size",               double_t,  0,     "Downsampling voxel size",                           0.1,        0.01,    1.0)
gen.add("num_normal_neighbors",     int_t,     0,     "Number of neighbors to search to estimate normals", 50,         5,       100)
gen.add("ground_normal_angle",      int_t,     0,     "angle of ground normal  ",                          30,         5,       100)
gen.add("cluster_tol",              double_t,  0,     "Distance to include point in cluster",              0.5,        0.001,   10.0)
gen.add("min_cluster_size",         int_t,     0,     "Minimum allowed number of points in a cluster",     12,          1,       1000)
gen.add("max_cluster_size",         int_t,     0,     "Maximum allowed number of points in a cluster",     1600,        100,     10000)

gen.add("ekf_speed_limit",  double_t,  0,     "set threshold for ekf speed",                2.0,        0.1,    10.0)
gen.add("max_match_dist",   double_t,  0,     "Maximum distance to associate measurement",  3.0,        0.1,    10.0)
gen.add("min_size_z",       double_t,  0,     "Minimum z dimension to track",               0.3,        0.1,    20.0)
gen.add("y_window",         double_t,  0,     "Lateral distance of tracking window",        10.0,       0.1,    200.0)
gen.add("min_age",          double_t,  0,     "Minimum age for output",                     1.0,        0.0,    5.0)
gen.add("q_pos",            double_t,  0,     "Kalman state standard deviation",            3.0,        1e-6,   100.0)
gen.add("q_vel",            double_t,  0,     "Kalman state standard deviation",            5.0,        1e-6,   100.0)
gen.add("r_pos",            double_t,  0,     "Kalman measurement standard deviation",      0.5,        1e-6,   100.0)

exit(gen.generate(PACKAGE, PACKAGE, "Lead_follow"))
